ci-operator/step-registry/openstack/provision/machinesubnet/openstack-provision-machinesubnet-commands.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+
+# This script will create a network, subnet, router, then plug the subnet into that network
+# and connect the router to the external networ.
+# The resources UUIDs are written in SHARED_DIR.
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+if [[ "${CONFIG_TYPE}" != "fipless" ]]; then
+    echo "Skipping step due to CONFIG_TYPE not being fipless."
+    exit 0
+fi
+
+export OS_CLIENT_CONFIG_FILE=${SHARED_DIR}/clouds.yaml
+CLUSTER_NAME=$(<"${SHARED_DIR}"/CLUSTER_NAME)
+OPENSTACK_EXTERNAL_NETWORK="${OPENSTACK_EXTERNAL_NETWORK:-$(<"${SHARED_DIR}/OPENSTACK_EXTERNAL_NETWORK")}"
+
+NET_ID=$(openstack network create "${CLUSTER_NAME}-network" | grep " id " | awk '{ print $4 }')
+echo "Created network: ${NET_ID}"
+echo $NET_ID>${SHARED_DIR}/MACHINESSUBNET_NET_ID
+
+SUBNET_ID=$(openstack subnet create "${CLUSTER_NAME}-subnet" \
+          --network ${NET_ID} --subnet-range ${SUBNET_RANGE} \
+          --dns-nameserver ${DNS_IP} \
+          --allocation-pool start=${ALLOCATION_POOL_START},end=${ALLOCATION_POOL_END} \
+          | grep " id " | awk '{ print $4 }')
+echo "Created subnet: ${SUBNET_ID}"
+echo ${SUBNET_ID}>${SHARED_DIR}/MACHINESSUBNET_SUBNET_ID
+echo ${SUBNET_RANGE}>${SHARED_DIR}/MACHINESSUBNET_SUBNET_RANGE
+echo ${API_VIP}>${SHARED_DIR}/API_VIP
+echo ${INGRESS_VIP}>${SHARED_DIR}/INGRESS_VIP
+
+ROUTER_ID=$(openstack router create "${CLUSTER_NAME}-router" | grep " id " | awk '{ print $4 }')
```

Comment:
nit: you can fine tune the command output without the need for grep/awk with `-f value -c id` (see above)